02-oop-inheritance/3. Working with classes/main.js

```diff
@@ -0,0 +1,89 @@
+class EventEmitter{
+    constructor(){
+        this.events = {};
+    }
+
+    on(eventName, callback){
+        if(!this.events[eventName]){
+            this.events[eventName] = [];
+        }
+              
+        this.events[eventName].push(callback);
+    }
+
+    emit(eventName){    
+        if(this.events[eventName]){
+            this.events[eventName].forEach(callback => {
+                callback.call(eventName);
+            });
+        }
+    }
+
+    off(eventName, _callback){
+        this.events[eventName] = this.events[eventName].filter(callback => callback !== _callback);
+    }
+}
+
+class Movie extends EventEmitter{
+    constructor(title, year, duration){
+        super();
+        this.title = title;
+        this.year = year;
+        this.duration = duration;
+        this.cast = [];
+    }
+    play(){
+        super.emit('play');
```

Comment:
No hace falta hacer referencia a `super`, utilizando `this` alcanza ya que inicialmente va a buscar el metodo `emit` dentro de Movie y cuando no lo encuentre va a recurrir a la clase padre la cual es EventEmitter que en este caso contiene al metodo